### PR TITLE
Bugfix report different behavior based on path

### LINKED ISSUE #257: Diffrent behavoiur with different abs/relative arguments
It is quite disappointing when overall project, separate file and folder checks give different results. Here is a little table showing results depended on arguments given.

<table border=1>
  <thead>
<tr>
 <td rowspan=3><b>cmd</td>
 <td rowspan=3><b>--absolute-path</td>
 <td colspan=7><b>PATH</td>
 <td colspan=3 rowspan="2"><b>results</td>
</tr>
<tr>
 <td rowspan=2><b>not defined</td>
 <td rowspan=2><b>Full path to project</td>
 <td rowspan=2><b>.</td>
 <td colspan=2><b>file</td>
 <td colspan=2><b>folder</td>
</tr>
<tr>
 <td><b>relative</td>
 <td><b>abs</td>
 <td><b>relative</td>
 <td><b>abs</td>
 <td><b>code</td>
 <td><b>import-error</td>
 <td><b>ignored noqa</td>
</tr>
  </thead>
  
<tr><td>  </td>
  <td>-</td><td>*</td><td> </td><td> </td><td> </td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>
  
<tr><td>   `pwd`</td>
  <td>-</td><td> </td><td>*</td><td> </td><td> </td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>

<tr><td>   .</td>
  <td>-</td><td> </td><td> </td><td>*</td><td> </td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>

<tr><td>   lib/__init__.py</td>
  <td>-</td><td> </td><td> </td><td> </td><td>*</td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>

<tr><td>   `pwd`/lib/__init__.py</td>
  <td>-</td><td> </td><td> </td><td> </td><td> </td><td>*</td><td> </td><td> </td><td>fail</td><td> </td><td>*</td></tr>

<tr><td>   lib</td>
  <td>-</td><td> </td><td> </td><td> </td><td> </td><td> </td><td>*</td><td> </td><td>fail</td><td>*</td><td>*</td></tr>

<tr><td>   `pwd`/lib</td>
  <td>-</td><td> </td><td> </td><td> </td><td> </td><td> </td><td> </td><td>*</td><td>fail</td><td>*</td><td>*</td></tr>

<tr><td>   --absolute-path</td>
  <td>+</td><td>*</td><td> </td><td> </td><td> </td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>

<tr><td>   --absolute-path `pwd`</td>
  <td>+</td><td> </td><td>*</td><td> </td><td> </td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>

<tr><td>   --absolute-path .</td>
  <td>+</td><td> </td><td> </td><td>*</td><td> </td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>

<tr><td>   --absolute-path lib/__init__.py</td>
  <td>+</td><td> </td><td> </td><td> </td><td>*</td><td> </td><td> </td><td> </td><td>ok  </td><td> </td><td> </td></tr>

<tr><td>   --absolute-path `pwd`/lib/__init__.py</td>
  <td>+</td><td> </td><td> </td><td> </td><td> </td><td>*</td><td> </td><td> </td><td>ok</td><td> </td><td> </td></tr>

<tr><td>   --absolute-path lib</td>
  <td>+</td><td> </td><td> </td><td> </td><td> </td><td> </td><td>*</td><td> </td><td>fail</td><td>*</td><td>*</td></tr>

<tr><td>   --absolute-path `pwd`/lib</td>
  <td>+</td><td> </td><td> </td><td> </td><td> </td><td> </td><td> </td><td>*</td><td>fail</td><td>*</td><td>*</td></tr>

</table>

### PR DESCRIPTION
**Pull request by chocoelho**
Resolves #257.

@carlio this will really need your input on it. To summarize, that was a point where `prospector` would give different results, this should be fixed. Also, `pep8linter` was not using all found files depending on the path it was called.

### CODE REVIEW

**Comment by carlio** [comment]
@chocoelho are you using an automatic formatter? Most of the changes are cosmetic which makes it hard to figure out what actual functionality has changed.

I'm all for having a conistent coding style - I'm a big fan of [black](https://github.com/ambv/black) and there's a pre-commit hook to add to `.pre-commit.yml` if we want to. I just think that either we don't use one or we use the same one to avoid too much noise in PRs.

**Comment by chocoelho** [comment]
I've been using YAPF, but I can switch to black if that's preferrable. I've set my editor to format as I touch files, this can be turned off for prospector as well.

Carlos Coêlho
Consultant & Developer

carlospecter@gmail.com (https://link.getmailspring.com/link/1532365759.local-314e4f8c-5149-v1.3.0-fd741eb7@getmailspring.com/0?redirect=mailto%3Acarlospecter%40gmail.com&recipient=cmVwbHkrMDAzNGNkNTMxOTk1M2RiMzg2Njc0NjY4YzkwMmYzMmYwYjMxOTg3YjcxNTU1MzliOTJjZjAwMDAwMDAxMTc2ZGM4YmU5MmExNjljZTE0N2I3ODlmQHJlcGx5LmdpdGh1Yi5jb20%3D)
https://github.com/chocoelho (https://link.getmailspring.com/link/1532365759.local-314e4f8c-5149-v1.3.0-fd741eb7@getmailspring.com/3?redirect=https%3A%2F%2Fgithub.com%2Fchocoelho&recipient=cmVwbHkrMDAzNGNkNTMxOTk1M2RiMzg2Njc0NjY4YzkwMmYzMmYwYjMxOTg3YjcxNTU1MzliOTJjZjAwMDAwMDAxMTc2ZGM4YmU5MmExNjljZTE0N2I3ODlmQHJlcGx5LmdpdGh1Yi5jb20%3D)

File: prospector/finder.py [hunk 1]
```diff
@@ -187,7 +188,7 @@ def _find_paths(ignore, curpath, rootpath):
     return files, modules, packages, directories
 
 
-def find_python(ignores, paths, explicit_file_mode, workdir=None):
```
Comment (chocoelho) [inline prospector/finder.py]: check here

File: prospector/finder.py [hunk 2]
```diff
@@ -198,5 +199,5 @@ def find_python(ignores, paths, explicit_file_mode, workdir=None):
         return SingleFiles(paths, workdir or os.getcwd())
     else:
         assert len(paths) == 1
-        files, modules, directories, packages = _find_paths(ignores, paths[0], paths[0])
```
Comment (chocoelho) [inline prospector/finder.py]: check here

**Comment by carlio** [comment]
I think it'd be best to avoid it because you can't guarantee every person is using the same one and PRs would get very difficult to keep track of. So if you could turn it off then create a PR with just the functional changes that'd be great - otherwise any PR from anyone would conflict in annoying ways!

File: prospector/tools/pep8/__init__.py [hunk 3]
```diff
@@ -116,25 +118,56 @@ def configure(self, prospector_config, found_files):
                     # this file exists - but does it have pep8 config in it?
                     header = re.compile(r'\[pep8\]')
                     with open(conf_path) as conf_file:
-                        if any([header.search(line) for line in conf_file.readlines()]):
+                        if any([header.search(line)
+                                for line in conf_file.readlines()]):
                             external_config = conf_path
                             break
 
+        # create a list of packages, but don't include packages which are
```
Comment (chocoelho) [inline prospector/tools/pep8/__init__.py]: check here

**Comment by chocoelho** [comment]
I'll rebase it with what indeed changed. I touched a some other files while debugging some parts of the code.

**Comment by chocoelho** [comment]
@carlio updated the PR the needed changes

**Pull request merged**